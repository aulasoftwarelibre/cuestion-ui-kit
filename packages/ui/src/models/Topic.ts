export interface Topic {
  label: string;
}

export default Topic;
