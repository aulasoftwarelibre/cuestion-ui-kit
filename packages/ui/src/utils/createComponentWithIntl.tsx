import { loadTranslationObject } from "enzyme-react-intl";
import * as React from "react";
import { IntlProvider } from "react-intl";
import * as renderer from "react-test-renderer";

import { languages } from "@cuestion/common";

export const createComponentWithIntl = (children, props = { locale: "en" }) => {
  return renderer.create(<IntlProvider {...props}>{children}</IntlProvider>);
};

loadTranslationObject(languages.en);

export { mountWithIntl, shallowWithIntl } from "enzyme-react-intl";
