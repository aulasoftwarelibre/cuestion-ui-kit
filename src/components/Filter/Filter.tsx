import * as React from "react";
import { createStyles, makeStyles, Theme } from '@material-ui/core/styles';
import Chip from '@material-ui/core/Chip';
import { InjectedIntl, injectIntl } from "react-intl";
import DoneIcon from '@material-ui/icons/Done';

export interface Props {
  onChangeHandler: any;
  topics: any;
  intl: InjectedIntl;
}

const topicos = [
  'Oliver Hansen',
  'Van Henry',
  'April Tucker',
  'Ralph Hubbard',
  'Omar Alexander',
  'Carlos Abbott',
  'Miriam Wagner',
  'Bradley Wilkerson',
  'Virginia Andrews',
  'Kelly Snyder',
];

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    root: {
      display: 'flex',
      justifyContent: 'center',
      flexWrap: 'wrap',
    },
    chip: {
      margin: theme.spacing(1),
    },
  }),
);

const handleDelete = () => {
  alert("you've clicked");
}

const _Filter: React.FunctionComponent<Props> = ({ onChangeHandler, topics }) => {
  const classes = useStyles();

  return (
    <div className={classes.root}>
      {topicos.map(topic => (
        <Chip
          label={topic}
          color="primary" 
          deleteIcon={<DoneIcon />}
          onDelete={handleDelete}
        />
      ))}
    </div>
  );
};

export const Filter = injectIntl(_Filter);
export default Filter;

