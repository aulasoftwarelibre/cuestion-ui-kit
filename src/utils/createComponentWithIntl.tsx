import * as React from "react";
import { IntlProvider } from "react-intl";
import * as renderer from "react-test-renderer";

const createComponentWithIntl = (children, props = { locale: "en" }) => {
  return renderer.create(<IntlProvider {...props}>{children}</IntlProvider>);
};

export default createComponentWithIntl;
