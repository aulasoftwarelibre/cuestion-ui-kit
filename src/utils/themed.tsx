import * as React from "react";
import * as renderer from "react-test-renderer";
import { ThemeProvider } from "styled-components";
import theme from "../theme";

const themed = (component: React.ReactElement) =>
  renderer.create(<ThemeProvider theme={theme}>{component}</ThemeProvider>);

export default themed;
