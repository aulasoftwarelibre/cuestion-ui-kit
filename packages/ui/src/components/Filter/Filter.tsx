import * as React from "react";

import Chip from "@material-ui/core/Chip";
import { createStyles, makeStyles, Theme } from "@material-ui/core/styles";
import DoneIcon from "@material-ui/icons/Done";

import { InjectedIntl, injectIntl } from "react-intl";
import { Topic } from "../../models/Topic";

export interface Props {
  onChangeHandler: any;
  topics: string[];
  intl: InjectedIntl;
}

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    root: {
      display: "flex",
      justifyContent: "center",
      flexWrap: "wrap",
    },
    chip: {
      margin: theme.spacing(1),
    },
  }),
);

const _Filter: React.FunctionComponent<Props> = ({
  onChangeHandler,
  topics,
}) => {
  const classes = useStyles();

  const [selected, setSelected] = React.useState<string[]>([]);

  const onClickHandler = (topic: string) => (
    event: React.MouseEvent<HTMLDivElement, MouseEvent>,
  ) => {
    const index: number = selected.indexOf(topic);
    const updatedSelected: string[] =
      index >= 0
        ? [...selected.slice(0, index), ...selected.slice(index + 1)]
        : [...selected, topic];

    setSelected(updatedSelected);
    onChangeHandler(updatedSelected);
  };

  return (
    <div className={classes.root}>
      {topics.map(topic => (
        <Chip
          key={topic}
          className={classes.chip}
          label={topic}
          onClick={onClickHandler(topic)}
          icon={selected.indexOf(topic) >= 0 ? <DoneIcon /> : undefined}
        />
      ))}
    </div>
  );
};

export const Filter = injectIntl(_Filter);
export default Filter;
