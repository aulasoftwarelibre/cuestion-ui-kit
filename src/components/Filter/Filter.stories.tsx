import { action } from "@storybook/addon-actions";
import { object, withKnobs } from "@storybook/addon-knobs";
import { storiesOf } from "@storybook/react";
import * as React from "react";

import { wInfo } from "../../utils/wInfo";
import { topics } from "../../models/Topic.mock";
import Filter from "./Filter";

const stories = storiesOf("Components/Filter", module);
stories.addDecorator(withKnobs);

stories.add(
  "default",
  wInfo()(() => <Filter topics={object("Topics", topics)} onChangeHandler={action("selected-topics")} />)
);
