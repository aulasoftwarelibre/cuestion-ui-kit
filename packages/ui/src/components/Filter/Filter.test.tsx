import Chip from "@material-ui/core/Chip";

import * as React from "react";

import { topics } from "../../models/Topic.mock";
import { createComponentWithIntl, mountWithIntl } from "../../utils/createComponentWithIntl";
import Filter from "./Filter";

describe("Filter", () => {
  const mockOnHandleClick = jest.fn();

  it("should render correctly", () => {
    const component = createComponentWithIntl(<Filter topics={topics} onChangeHandler={mockOnHandleClick} />).toJSON();

    expect(component).toMatchSnapshot();
  });

  it("should appear not selected by default", () => {
    const wrapper = mountWithIntl(<Filter topics={topics} onChangeHandler={mockOnHandleClick} />);

    expect(
      wrapper
        .find(Chip)
        .first()
        .prop("icon")
    ).toBeUndefined();
  });

  it("should appear selected", () => {
    const wrapper = mountWithIntl(<Filter topics={topics} onChangeHandler={mockOnHandleClick} />);

    wrapper
      .find(Chip)
      .first()
      .simulate("click");

    expect(
      wrapper
        .find(Chip)
        .first()
        .prop("icon")
    ).toBeDefined();
  });

  it("should return an array with the topic selected", () => {
    const wrapper = mountWithIntl(<Filter topics={topics} onChangeHandler={mockOnHandleClick} />);

    wrapper
      .find(Chip)
      .first()
      .simulate("click");

    expect(mockOnHandleClick).toBeCalledWith([topics[0]]);
  });

  it("should return an array without the topic deselected", () => {
    const wrapper = mountWithIntl(<Filter topics={topics} onChangeHandler={mockOnHandleClick} />);

    wrapper
      .find(Chip)
      .first()
      .simulate("click");
    wrapper
      .find(Chip)
      .first()
      .simulate("click");

    expect(mockOnHandleClick).lastCalledWith([]);
  });
});
