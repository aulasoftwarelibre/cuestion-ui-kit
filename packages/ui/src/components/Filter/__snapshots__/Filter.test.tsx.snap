// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter should render correctly 1`] = `
<div
  className="makeStyles-root-1"
>
  <div
    className="MuiChip-root makeStyles-chip-2 MuiChip-clickable"
    onClick={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    role="button"
    tabIndex={0}
  >
    <span
      className="MuiChip-label"
    >
      Agile
    </span>
  </div>
  <div
    className="MuiChip-root makeStyles-chip-2 MuiChip-clickable"
    onClick={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    role="button"
    tabIndex={0}
  >
    <span
      className="MuiChip-label"
    >
      Backend
    </span>
  </div>
  <div
    className="MuiChip-root makeStyles-chip-2 MuiChip-clickable"
    onClick={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    role="button"
    tabIndex={0}
  >
    <span
      className="MuiChip-label"
    >
      Frontend
    </span>
  </div>
</div>
`;
